File: tests/fixtures/code/typescript/sample.ts
/**
 * Sample TypeScript file for C3c fixture.
 *
 * Used by tests/test_code_enricher.py to exercise TypeScript scope-table building.
 */

/** Top-level exported function. */
export function topFn(x: number): number {
  return x * 2;
}

/** Class with a method. */
export class MyClass {
  private value: string;

  constructor(value: string) {
    this.value = value;
  }

  myMethod(): string {
    return this.value.toUpperCase();
  }
}

// Exported const arrow function assigned to a name.
// Arrow functions are treated as enclosing-scope (module-level) code in C3c v1.
export const arrowFn = (): void => {
  console.log("arrow");
};

File: tests/fixtures/code/python/sample.py
"""Sample module docstring for C3c fixture.

Used by tests/test_code_enricher.py to exercise scope-table building.
"""


def top_fn():
    """Top-level function."""
    return 42


class Outer:
    """Outer class with a nested class."""

    class_attr = "outer"

    def outer_method(self):
        """Method on Outer."""
        return self.class_attr

    class Inner:
        """Nested class inside Outer."""

        def inner_method(self):
            """Method on Inner (innermost class wins)."""
            return "inner"


# Module-level statement after Outer (provides a 'module' chunk)
MODULE_CONSTANT = "module_level"


def some_decorator(fn):
    """Simple decorator used below."""
    return fn


@some_decorator
def decorated_fn():
    """Function with a decorator."""
    return "decorated"
